OutOfSchool/OutOfSchool.WebApi/Services/PhotoStorage/PhotoStorage.cs

```diff
@@ -0,0 +1,350 @@
+                throw;
+            }
+        }
+
+        /// <inheritdoc/>
+        public async Task DeleteFiles(List<string> filesPaths)
+        {
+            try
+            {
+                logger.Information($"Process of deleting photos started.");
+
+                if (filesPaths is null)
+                {
+                    throw new ArgumentNullException(localizer["File paths can not be null!"]);
+                }
+
+                foreach (var path in filesPaths)
+                {
+                    File.Delete(path);
+
+                    Expression<Func<Photo, bool>> filter = p => p.Path == path;
+
+                    var photos = await repository.GetByFilter(filter).ConfigureAwait(false);
+
+                    await repository.Delete(photos.FirstOrDefault()).ConfigureAwait(false);
+                }
+
+                logger.Information($"Photo deleted photo successfully.");
+            }
+            catch (ArgumentException ex)
+            {
+                logger.Error($"Process of deleting photos failed. {ex}");
+                throw;
+            }
+            catch (IOException ex)
+            {
+                logger.Error($"Process of deleting photos failed. {ex}");
+                throw;
+            }
+        }
+
+        /// <inheritdoc/>
+        public async Task<List<byte[]>> GetFiles(long entityId, EntityType entityType)
+        {
+            try
+            {
+                logger.Information($"Process of getting photos started.");
+
+                var photos = new List<byte[]>();
+
+                Expression<Func<Photo, bool>> filter = p => p.EntityId == entityId && p.EntityType == entityType;
+
+                var images = await repository.GetByFilter(filter).ConfigureAwait(false);
+
+                foreach (var image in images)
+                {
+                    using (var stream = File.Open(image.Path, FileMode.Open))
+                    {
+                        byte[] fileContent;
+
+                        using (var ms = new MemoryStream())
+                        {
+                            await stream.CopyToAsync(ms).ConfigureAwait(false);
+
+                            fileContent = ms.ToArray();
+
+                            photos.Add(fileContent);
+                        }
+                    }
+                }
+
+                logger.Information($"Successfully got photos.");
+
+                return photos;
+            }
+            catch (ArgumentException ex)
+            {
+                logger.Error($"Process of getting photos failed. {ex}");
+                throw;
+            }
+            catch (IOException ex)
+            {
+                logger.Error($"Process of getting photos failed. {ex}");
+                throw;
+            }
+        }
+
+        /// <inheritdoc/>
+        public async Task<List<string>> GetFilesPaths(long entityId, EntityType entityType)
```

Comment:
GetFilesPaths and GetFilePath methods have some code duplications, mb we can extract some code into method that will be reused?